src/Tokenizer/Tokens.php

```diff
@@ -158,19 +160,23 @@ public static function detectBlockType(Token $token): ?array
      * Create token collection from array.
      *
      * @param array<int, Token> $array       the array to import
-     * @param ?bool             $saveIndices save the numeric indices used in the original array, default is yes
+     * @param bool              $saveIndices
      */
-    public static function fromArray($array, $saveIndices = null): self
+    public static function fromArray($array, $saveIndices = false): self
```

Comment:
master says:
> default is _YES_

PR says:
> default is _NO_

Reply:
How would you like to deprecate default?

Reply:
you shall not change the behaviour when introducing deprecation.
change of behaviour is not creating deprecation, it's already creating BC breaks.

keep same behaviour but trigger the msg only.
not ideal, I know, yet otherwise you change the behaviour

Reply:
I hope now everyone is happy :)

Reply:
"typehinted" means PHPDoc-umented, "typed" is maybe you want, but this is really monitored by PHPStan and you should be aware if you change...